Source/WebKit/UIProcess/Automation/WebAutomationSession.cpp

```diff
@@ -1241,6 +1274,12 @@ CommandResult<void> WebAutomationSession::dismissCurrentJavaScriptDialog(const I
     bool isShowingJavaScriptDialog = m_client->isShowingJavaScriptDialogOnPage(*this, *page);
     SYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!isShowingJavaScriptDialog, NoJavaScriptDialog);
 
+#if ENABLE(WEBDRIVER_BIDI)
+    auto apiDialogType = m_client->typeOfCurrentJavaScriptDialogOnPage(*this, *page);
+    SYNC_FAIL_WITH_PREDEFINED_ERROR_IF(!apiDialogType, InternalError);
```

Comment:
NIT: should this still fail even if the client is not using WebDriver BiDi (i.e. WebDriver "classic")?